docs/fastdds/transport/transport_api.rst

```diff
@@ -0,0 +1,126 @@
+
+The :ref:`transport_transportApi_transport` defines the following data members:
+
++-------------------------+------------------+-----------------------------------------------------------------------+
+| Member                  | Data type        | Description                                                           |
++=========================+==================+=======================================================================+
+| ``transport_kind_``     | ``int32_t``      | Unique identifier of the transport type.                              |
++-------------------------+------------------+-----------------------------------------------------------------------+
+
+
+.. _transport_transportApi_locator:
+
+Locator
+-------
+
+A :class:`Locator` uniquely identifies a communication channel with a remote peer for a particular transport.
+For example, on UDP transports, the :class:`Locator` will contain the information of the IP address and port
+of the remote peer.
+
+The :class:`Locator` class is not abstract, and no specializations are implemented for each transport type.
+Instead, transports should map the data members of the :class:`Locator` class to their own channel identification
+concepts. For example, on :ref:`transport_sharedMemory_sharedMemory` the ``address`` contains a unique ID
+for the local host, and the ``port`` represents the shared ring buffer used to communicate buffer descriptors.
+
+On :ref:`listening_locators` you can find more information about how to configure :ref:`dds_layer_domainParticipant`
+to listen to incoming traffic.
+
+Data members
+^^^^^^^^^^^^
+
+The :ref:`transport_transportApi_locator` defines the following data members:
+
++--------------+------------------+-----------------------------------------------------------------------+
+| Member       | Data type        | Description                                                           |
++==============+==================+=======================================================================+
+| ``kind``     | ``int32_t``      | Unique identifier of the transport type.                              |
++--------------+------------------+-----------------------------------------------------------------------+
+| ``port``     | ``uint32_t``     | The channel *port*.                                                   |
++--------------+------------------+-----------------------------------------------------------------------+
+| ``address``  | ``octet[16]``    | The channel *address*.                                                |
++--------------+------------------+-----------------------------------------------------------------------+
+
+In TCP, the port of the locator is divided into a physical and a logical port.
+The *physical port* is the port used by the network device, the real port that the operating system understands.
+The *logical port* can be seen as RTPS port, or UDP's equivalent port (physical ports of UDP, are logical ports in TCP).
```

Comment:
Both. Logical port is stored in the most significant word, and the physical in the least significant word.
Otherwise, a uint16 would suffice.

But explaining this on the docs seemed a bit too much. I would even remove the note about physical and logical ports, but there is no 'setPort' function on IPLocator, only 'setPhysical', 'setLogical' and 'setRTPS' ports, so something needs to be said

Reply:
Tried to clarify a little, but without deeper knowledge it will still be somehow confusing